### ISSUE TITLE
Renaming Doesn’t Work

### ISSUE THREAD

**Issue by Detonator-Prime** [MLDEG/AllMobHeads#16]
### Datapack version

10.4

### Description

Firstly, I absolutely love the pack. It has given my wife and I hours of enjoyment trying to get every head. Unfortunately, using a name tag to rename mobs as per the instructions is not working. Nothing happens. For example, renaming a chicken “Potato” doesn’t given me the head like it is supposed to. Same with trying to summon an Illusioner. Nothing happens.

### Steps to reproduce

When renaming 

### Expected behavior

Should drop head

**Comment by MLDEG** [comment]
Hi ! Thanks =)

Are you using a paper server, or something else?
Or are you on a vanilla server?

And finally, what version of minecraft is your server on? (I have not yet tested on version 1.21.4+)

Anyway I'll try to take a closer look tomorrow and see if I have anything to fix while waiting for your info :)

**Comment by Detonator-Prime** [comment]
Hello, thanks for the quick response. I am using a fabric server running game version 1.21.1.

**Comment by MLDEG** [comment]
Hi, sorry for the delay !
So, I test my datapack in 1.21.1 and the renaming works correctly.
Potentially a conflict with Fabric but I have no idea where that could come from.
![image](https://github.com/user-attachments/assets/481b91f7-d692-4880-b99f-23ee2bd446af)
Same with the Illusioner :/
![image](https://github.com/user-attachments/assets/c9956045-965d-4e63-9861-cf841c63ab84)
On the other hand, it doesn't work on the last snapshot which adds the new pigs, I'm working on it!

**Comment by Detonator-Prime** [comment]
Ok no worries. 
Thanks for checking.
Must be a conflict somewhere with my other mods.
It is weird because the JEB_ sheep worked but the randomly stopped working yesterday when my wife tried to get one. 
Just another quick question. Is it possible to reset something so you get another gold head for completing a set of villagers if you lose it? 
I got a set of villagers but my inventory was full so I didn’t see it on the ground before I left the village lol. 

Sent from Yahoo Mail for iPhone


On Friday, January 10, 2025, 11:32 am, MLDEG ***@***.***> wrote:




Hi, sorry for the delay !
So, I test my datapack in 1.21.1 and the renaming works correctly.
Potentially a conflict with Fabric but I have no idea where that could come from.
image.png (view on web)
Same with the Illusioner :/
image.png (view on web)
On the other hand, it doesn't work on the last snapshot which adds the new pigs, I'm working on it!

**Comment by MLDEG** [comment]
Yes totally, you can reset a head that you already had which completes a collection, then recover it to revalidate the collection :)
/advancement revoke "YOUR NAME" only "the advancement" <= (all_mob_heads:collection/villagers/desert/12_desert_villager, for example)

By the way, I updated the datapack to add the two new pig variant =D

**Comment by Detonator-Prime** [comment]
Thanks so much. I really appreciate you replying and for making such an awesome mod.

Sent from Yahoo Mail for iPhone


On Friday, January 10, 2025, 2:52 pm, MLDEG ***@***.***> wrote:




Yes totally, you can reset a head that you already had which completes a collection, then recover it to revalidate the collection :)
/advancement revoke "YOUR NAME" only "the advancement" <= (all_mob_heads:collection/villagers/desert/12_desert_villager, for example)

By the way, I updated the datapack to add the two new pig variant =D

**Comment by Detonator-Prime** [comment]
Hello again, 
Just wanted to let you know that I discovered what the issue was and it is completely on my end. 
After getting the jeb sheep to work I changed resource packs and part of the pack changed all of the font in MC to upper case. 
Hence when you look at the advancements the name tag looked like it had to be all uppercase and then didn’t work. 
As soon as I worked this out and made the name tags lowercase it worked like a charm lol. 
I did notice one thing though. I updated the data pack and now, duplicate heads do not stack on heads that were already collected. It is like the previous heads and the new head of the same type are being treated as different. 


Sent from Yahoo Mail for iPhone


On Friday, January 10, 2025, 10:07 pm, Yahoo Mail ***@***.***> wrote:

Thanks so much. I really appreciate you replying and for making such an awesome mod.

Sent from Yahoo Mail for iPhone


On Friday, January 10, 2025, 2:52 pm, MLDEG ***@***.***> wrote:




Yes totally, you can reset a head that you already had which completes a collection, then recover it to revalidate the collection :)
/advancement revoke "YOUR NAME" only "the advancement" <= (all_mob_heads:collection/villagers/desert/12_desert_villager, for example)

By the way, I updated the datapack to add the two new pig variant =D

**Comment by MLDEG** [comment]
Hi
Concerning the heads which don't stack, I think it's because Mojang has changed the behaviour of "components", previously named NBT, but not really sure about that.

And no worries about the renaming, I still looked to be sure and in any case I still had to change certain things regarding the name tags with the last snapshot (Mojang has changes, once again, how the custom names works.)
